pkg/controller/atlascluster/cluster.go

```diff
@@ -0,0 +1,59 @@
+package atlascluster
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	mdbv1 "github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/controller/atlas"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/controller/workflow"
+	"go.mongodb.org/atlas/mongodbatlas"
+)
+
+func ensureClusterState(ctx context.Context, wctx *workflow.Context, connection atlas.Connection, project *mdbv1.AtlasProject, cluster *mdbv1.AtlasCluster) (c *mongodbatlas.Cluster, _ workflow.Result) {
```

Comment:
I see you pass the `ctx` from outside.. 
This makes sense (even though we use the noop `background`) - but do you think we could encapsulate it into our `workflow.Context`?
The other alternative would be just to pass `ctx.Background` to `client.Clusters.Get(` as we don't use that functionality)

What do you think?

Reply:
Also is there any special reason to return the pointer to `mongodbatlas.Cluster` instead of a copy?

Reply:
Passing `Context` is a habit at this point 😃 
I'll change it to `Background` since we're probably not going to use it

Returning `*Cluster` was done for consistency, as it is returned via pointer everywhere else - no hard reasons.